### PR TITLE
fix: destroyed stream manager can be retried

### PR DESCRIPTION
**Pull request by ThomasRalee**
## Summary by CodeRabbit

- **New Features**
  - Added a terminal **Destroyed** stream state.
- **Bug Fixes**
  - Streams can no longer start/connect/retry after being destroyed, including during in-flight reconnect handling.
  - Destruction now performs immediate cleanup, cancels pending retries, and emits a final disconnect with `willRetry: false` (once), transitioning to **Destroyed**.
  - Repeated `destroy()` calls are safe no-ops; teardown is more robust—unsubscribe abort errors are suppressed, other failures emit warnings, and disconnect listener exceptions during destroy are handled cleanly.
- **Tests**
  - Expanded lifecycle and edge-case coverage for destruction semantics.

### CODE REVIEW

**Review by Frederick-88 — APPROVED** [review]

File: packages/sdk-ts/src/client/indexer/grpc_stream/streamV2/StreamManagerV2.ts [hunk 1]
```diff
@@ -166,29 +172,72 @@ export class StreamManagerV2<TResponse> extends EventEmitter<
...
       return
     }
 
-    this.subscription.unsubscribe()
+    const subscription = this.subscription
     this.subscription = null
+
+    try {
+      subscription.unsubscribe()
+    } catch (error) {
```
Comment (ivan-angjelkoski) [inline packages/sdk-ts/src/client/indexer/grpc_stream/streamV2/StreamManagerV2.ts:251]: is there a reason why we are extracting this to a const? in looks like we can use it directly in the try block:
```
try {
  this.subscription.unsubscribe()
}
```

and then in the finally block we can set it to null

**Review by ivan-angjelkoski — APPROVED** [review]
